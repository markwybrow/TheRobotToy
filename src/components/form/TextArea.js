import React from 'react';
import PropTypes from 'prop-types';
const InputTextArea = (props) => {
  const { placeholder = '', values, fnc = () => {} } = props;
  return (
    <div className="control">
      <textarea className="textarea" value={values} onChange={fnc(values)} placeholder={placeholder} />
    </div>
  );
};

InputTextArea.propTypes = {
  values: PropTypes.string,
  fnc: PropTypes.func,
  placeholder: PropTypes.string,
};
export default InputTextArea;
