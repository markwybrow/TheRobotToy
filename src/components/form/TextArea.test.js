import React from 'react';
import { shallow } from 'enzyme';
import TextArea from './TextArea';
describe('TextArea', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<TextArea debug />);

    expect(component).toMatchSnapshot();
  });
});
