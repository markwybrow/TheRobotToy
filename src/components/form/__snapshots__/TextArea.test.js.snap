// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="control"
>
  <textarea
    className="textarea"
    placeholder=""
  />
</div>
`;

exports[`TextArea should render correctly in "debug" mode 1`] = `
<div
  className="control"
>
  <textarea
    className="textarea"
    placeholder=""
  />
</div>
`;
